src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java

```diff
@@ -237,6 +237,8 @@ private void validateMetricTier(final Metric metric) throws QueryRejectedExcepti
 				ret.add(queryMetric);
 			} catch (ContextualJsonSyntaxException e) {
 				throw new BeanValidationException(new SimpleConstraintViolation(e.getContext(), e.getMessage()), context);
+			} catch (QueryRejectedException e) {
```

Comment:
Since we are suppressing the exception here and continue iterating over the `metricsArray`, I wonder if the resulting gap is not going to break rendering?

Consider a query of 3 metrics where first and last are not rejected, but the one in the middle of the list is.  We then calculate `context` as `query.metric[0]` and `query.metric[2]`, but we skip over `query.metric[1]`.
I'm not all that familiar with the KDB API, but I think we should verify what happens on the client (i.e. Grafana) once it gets fewer items than it expects in response.

I would prefer that we return the same number of items in response as we would if rejection was not activated (requested 3 metrics, returned 3 metrics), simply with the empty list of data points for the rejected ones.

Reply:
agreed! after quick discussion, this is what we need to - return empty list of datapoints